scripts/destroy_infrastructure.sh

```diff
@@ -1,13 +1,20 @@
 #!/usr/bin/env bash
 
-set -e
+set -euxo pipefail
 
 TERRAFORM_STATE_BUCKET="${TERRAFORM_STATE_BUCKET:-eq-terraform-load-generator-tfstate}"
 
-terraform init --upgrade --backend-config prefix=${PROJECT_NAME} -var "project_name=${PROJECT_NAME}" --backend-config bucket=${TERRAFORM_STATE_BUCKET}
+tfenv use "$(< .terraform-version)"
 
-# This will get destroyed as part of deleting the project
-terraform state rm google_compute_network.k8s
+terraform init --upgrade --backend-config prefix=${TF_VAR_project_name} --backend-config bucket=${TERRAFORM_STATE_BUCKET}
+
+# Do not delete the project or bucket.
+terraform state rm google_project.project
+terraform state rm google_storage_bucket.benchmark-output-storage
 
 # Destroy infrastructure
-terraform destroy -auto-approve -var "project_name=${PROJECT_NAME}"
\ No newline at end of file
+terraform destroy --auto-approve
+
+# Import project and bucket resouce back into the state
+terraform import google_project.project "$PROJECT_ID"
+terraform import google_storage_bucket.benchmark-output-storage ${PROJECT_ID}-benchmark-outputs
```

Comment:
If we always require a project to be created, then `PROJECT_ID` here can just be `TF_VAR_project_name`. It will also mean we don't need to run the `gcloud auth activate-service-account` command in the destroy infrastructure task.

Reply:
In that case, `TF_VAR_project_name` should potentially be called `TF_VAR_project_id`?

Reply:
Fair point. We should also remove the project resource from Terraform. Then we wouldn't need to manually remove it from the state and re-add it.

Reply:
I'm not sure what the impact of removing the project will be - don't resources such as compute require a project resource before terraform can provision them?

Reply:
I don't think so, they need a project id and we currently get that from the project resource, but if it was passed in we can just use a var.